src/Input/GamePadAction.cs

```diff
@@ -0,0 +1,381 @@
+                { "y", b => b.Buttons.Y },
+
+                { "start", b => b.Buttons.Start },
+                { "back", b => b.Buttons.Back },
+                { "bigbutton", b => b.Buttons.BigButton },
+
+                { "leftshoulder", b => b.Buttons.LeftShoulder },
+                { "leftstick", b => b.Buttons.LeftStick },
+                { "rightshoulder", b => b.Buttons.RightShoulder },
+                { "rightstick", b => b.Buttons.RightStick },
+
+                { "dpadright", b => b.DPad.Right },
+                { "dpadleft", b => b.DPad.Left },
+                { "dpadup", b => b.DPad.Up },
+                { "dpaddown", b => b.DPad.Down }
+            };
+
+            private readonly InputManager.GamePadStateContainer pad;
+            private readonly string buttonName;
+            private readonly ButtonSelector buttonSelector;
+
+            public GamePadButtonAction(int id, string buttonName, ButtonSelector buttonSelector)
+            {
+                this.pad = InputManager.GamePads[id];
+                this.buttonName = buttonName;
+                this.buttonSelector = buttonSelector;
+            }
+
+// ReSharper disable once MemberHidesStaticFromOuterClass
+            public static IEnumerable<GamePadButtonAction> GetAll(int id)
+            {
+                return GamePadButtonAction.buttonSelectors.Select(
+                    pair => new GamePadButtonAction(id, pair.Key, pair.Value)
+                    );
+            }
+
+
+            public static ButtonSelector GetButtonSelector(string buttonName)
+            {
+                ButtonSelector selector;
+
+                GamePadButtonAction.buttonSelectors.TryGetValue(buttonName, out selector);
+
+                return selector;
+            }
+
+            private bool downNow
+            {
+                get { return this.buttonSelector(this.pad.CurrentState) == ButtonState.Pressed; }
+            }
+
+            private bool downBefore
+            {
+                get { return this.buttonSelector(this.pad.PreviousState) == ButtonState.Pressed; }
+            }
+
+            public override bool Hit
+            {
+                get { return this.downNow && !this.downBefore; }
+            }
+
+            public override bool Active
+            {
+                get { return this.downNow; }
+            }
+
+            public override bool Released
+            {
+                get { return this.downBefore && !this.downNow; }
+            }
+
+            public override string ToString()
+            {
+                return "gamepad" + this.pad.Id + ":" + this.buttonName;
+            }
+
+        }
+
+        sealed private class GamePadAxisAction : IAction
+        {
+            private readonly static Dictionary<string, AxisSelector> axisSelectors = new Dictionary<string, AxisSelector>
+            {
+                { "+x", s => s.ThumbSticks.Left.X },
+                { "-x", s => -s.ThumbSticks.Left.X },
```

Comment:
Given that some controllers might not have two sticks, or might have the sticks reversed, or the x and y axis for some sticks reversed... I think in the least a lot more research and maybe a more flexible solution is needed.